### ISSUE TITLE
DataFrameWidget is wrong when several columns have the same name

### ISSUE THREAD

**Issue by dragos** [andypetrella/spark-notebook#612]
This is from our Spark training.
DataFrameWidget is wrong when two columns have the same name. For example, the following query has two fields called "airport", coming from the "airports" table, but with different aliases. However, the widget will show the same value in both columns, even though the data frame prints correctly if I do `df.show()`.

```
SELECT f.origin, a.airport, f.dest, b.airport, f.count
  FROM flights_between_airports2 f
  JOIN airports a ON f.origin = a.iata
  JOIN airports b ON f.dest = b.iata
```

I can work around this by adding aliases for “a.airport AS airport1” and “b.airport AS ..." in the SELECT:

```
SELECT f.origin, a.airport AS airport_a, f.dest, b.airport AS airport_b, f.count
  FROM flights_between_airports2 f
  JOIN airports a ON f.origin = a.iata
  JOIN airports b ON f.dest = b.iata
```

**Comment by andypetrella** [comment]
ahah.... well, besides the fact that this is probably also wrong :-), we need to find a way to use something else as the key for column names, which goes again relation algebra (two vars cannot have the same unless they have identical values ^^)

**Comment by vidma** [comment]
hmm when spark notices ambigous column names it prints them like `colName#77` vs `colName#33`, so we could use this as the key. didn't check yet how to get it, but using column index would work (besides complicating all the things, without much added value).

easier would be to transform the dataframe before displaying, instead of handling this extreme case in the widgets...
```scala
  df.withColumn($"$dupCol#$num", $"$someHowGetAliasOfColumn")
```

wdyt?

**Comment by andypetrella** [comment]
Yeah, what needs to be done then is to use this thing if the index as key and show only the name in the UI I guess

**Comment by vidma** [comment]
but only if it do not complicate things...

**Comment by vidma** [comment]
@dragos , btw why do you need `DataFrameWidget` at all (is it anyhow better than others)?
    - unlimited paging, oh well, who needs that?

there is whole bunch of widgets inside [front/widgets/package.scala](https://github.com/andypetrella/spark-notebook/blob/master/modules/common/src/main/scala/notebook/front/widgets/package.scala), including `TableChart(someDF, maxPoints = 20)`, `PivotChart`, `display` etc.
I'm considering even to drop/deprecate the `DataFrameWidget` until we refactor all other widgets to be `more-powerful`.
